src/utils/sweepline.jl

```diff
@@ -0,0 +1,182 @@
+      end
+    end
+    if !isnothing(prev)
+      newgeom, newtype = _newevent(Segment(BinaryTrees.key(prev)), s)
+      if _checkintersection(newtype)
+        BinaryTrees.insert!(𝒬, newgeom)
+        haskey(𝒞, newgeom) ? push!(𝒞[newgeom], BinaryTrees.key(prev), vertices(s)) :
+        (𝒞[newgeom] = [BinaryTrees.key(prev), vertices(s)])
+      end
+    end
+    if !isnothing(next)
+      newgeom, newtype = _newevent(s, Segment(BinaryTrees.key(next)))
+      if _checkintersection(newtype)
+        BinaryTrees.insert!(𝒬, newgeom)
+        haskey(𝒞, newgeom) ? push!(𝒞[newgeom], vertices(s), BinaryTrees.key(next)) :
+        (𝒞[newgeom] = [vertices(s), BinaryTrees.key(next)])
+      end
+    end
+  end
+end
+
+function _processend!(ℰ, 𝒬, 𝒯, 𝒞)
+  for s in ℰ
+    prev, next = BinaryTrees.prevnext(𝒯, s)
+    BinaryTrees.delete!(𝒯, s)
+    s = Segment(s)
+    if !isnothing(prev) && !isnothing(next)
+      newgeom, newtype = _newevent(Segment(BinaryTrees.key(next)), Segment(BinaryTrees.key(prev)))
+      if _checkintersection(newtype)
+        BinaryTrees.insert!(𝒬, newgeom)
+        haskey(𝒞, newgeom) ? push!(𝒞[newgeom], BinaryTrees.key(next), BinaryTrees.key(prev)) :
+        (𝒞[newgeom] = [BinaryTrees.key(next), BinaryTrees.key(prev)])
+      end
+    end
+  end
+end
+
+function _processintersects!(ℐ, 𝒬, 𝒯, 𝒞)
+  for s in ℐ
+    prev, _ = BinaryTrees.prevnext(𝒯, s)
+    if !isnothing(prev)
+
+      # Find segments r and u
+      r, _ = BinaryTrees.prevnext(𝒯, s)
+      _, u = BinaryTrees.prevnext(𝒯, BinaryTrees.key(prev))
+
+      # Remove crossing points rs and tu from event queue
+      if !isnothing(r)
+        newgeom, newtype = _newevent(Segment(BinaryTrees.key(r)), Segment(s))
+        if _checkintersection(newtype)
+          BinaryTrees.delete!(𝒬, newgeom)
+        end
+      end
+      if !isnothing(u)
+        newgeom, newtype = _newevent(Segment(BinaryTrees.key(u)), Segment(BinaryTrees.key(prev)))
+        if _checkintersection(newtype)
+          BinaryTrees.delete!(𝒬, newgeom)
+        end
+      end
+
+      # Add crossing points rt and su to event queue
+      if !isnothing(r)
+        newgeom, newtype = _newevent(Segment(BinaryTrees.key(r)), Segment(BinaryTrees.key(prev)))
+        if _checkintersection(newtype)
+          BinaryTrees.insert!(𝒬, newgeom)
+          haskey(𝒞, newgeom) ? push!(𝒞[newgeom], BinaryTrees.key(r), BinaryTrees.key(prev)) :
+          (𝒞[newgeom] = [BinaryTrees.key(r), BinaryTrees.key(prev)])
+        end
+      end
+      if !isnothing(u)
+        newgeom, newtype = _newevent(Segment(BinaryTrees.key(u)), Segment(s))
+        if _checkintersection(newtype)
+          BinaryTrees.insert!(𝒬, newgeom)
+          haskey(𝒞, newgeom) ? push!(𝒞[newgeom], BinaryTrees.key(u), s) : (𝒞[newgeom] = [BinaryTrees.key(u), s])
+        end
+      end
+    end
+  end
+end
+
+_pushintersection(lookup, segments) = unique([lookup[segment] for segment in segments])
```

Comment:
```suggestion
_pushintersection(lookup, segments) = unique(lookup[segment] for segment in segments)
```
The `unique` functions works with generators, in this case, it returns a `Vector`.